master/internal/api_shell.go

```diff
@@ -51,18 +52,16 @@ func (a *apiServer) GetShells(
 	if err = a.ask(shellsAddr, req, &resp); err != nil {
 		return nil, err
 	}
-
+	limitedScopes, err := command.AuthZProvider.Get().AccessibleScopes(
+		ctx, *curUser, model.AccessScopeID(req.WorkspaceId),
+	)
+	if err != nil {
+		return nil, apiutils.MapAndFilterErrors(err, nil, nil)
```

Comment:
- i don't have an intg test for the plural version of getNTSC
- are there cases we'd wanna swap the permission erro for 404?